python/tvm/relay/frontend/keras.py

```diff
@@ -653,12 +653,15 @@ def from_keras(model, shape=None):
         raise ValueError("Keras frontend currently supports data_format = channels_last only.")
     _check_unsupported_layers(model)
 
+    def _convert_input_layer(keras_layer):
+        input_name = keras_layer.name
+        input_shape = shape[input_name] if shape is not None and input_name in shape else None
+        etab.set_expr(input_name, new_var(input_name, shape=input_shape))
```

Comment:
I guess we also need to update set_expr so that we can call the method multiple times against the same input_name.
```diff
     def set_expr(self, name, expr):
         assert isinstance(expr, _expr.Expr)
-        self.exprs[name] = expr
+        if name not in self.exprs:
+            self.exprs[name] = expr
```

Reply:
👍 thanks